tagmanager/src/tm_source_file.c

```diff
@@ -109,6 +109,35 @@ gchar *tm_get_real_path(const gchar *file_name)
 	return NULL;
 }
 
+/* add argument list of __init__() Python methods to the class tag */
+static void update_python_arglist(const TMTag *tag)
```

Comment:
>Mmmwell, this is pretty specific to the Python parser and not well-contained, but well, probably still better than a whole new hook indeed.

I don't find it a big problem - Geany is full of if (language) {do stuff} things and this is just one more (and IMO much cleaner than the direct parser->TM callback).

> BTW, this might be fixable in the parser side using the U-CTags' new Cork thingie? Not sure and not tested, but that would be nice.

I haven't checked what the cork thing in uctags does so not sure either.